staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -813,6 +797,26 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 	return s.versioner.UpdateList(listObj, uint64(withRev), "", nil)
 }
 
+func (s *store) getList(ctx context.Context, keyPrefix string, recursive bool, options clientv3.ListOptions) (clientv3.KubernetesListResponse, error) {
```

Comment:
cc @wojtek-t 
For recursive implementation

Reply:
Long term I would want to extract stores `GetList(recursive=true)`  into pure `List()`. With the new interface the code will be much simpler.